File: Count the number of full binary tree/code.java
class Solution {
    public static int numoffbt(int arr[], int n) {
        int min = 100001, max = 1;

        for (int  i : arr) {
            max = Math.max(max, i);
            min = Math.min(min, i);
        }

        int[] dp = new int[max + 1];
        for (int i : arr) dp[i] = 1;

        int ans = 0;

        for (int i = min; i <= max; i++) {
            if (dp[i] == 0) continue;

            for (int j = i + i; j <= max && j / i <= i; j += i) {
                if (dp[j] == 0) continue;

                dp[j] += (dp[i] * dp[j / i]);
                if (i != j / i) dp[j] += (dp[i] * dp[j / i]);
            }

            ans = (ans + dp[i]) % 1000000007;
        }

        return ans % 1000000007;
    }
}

File: Rearrange array alternately/code.cpp
  void rearrange(long long *arr, int n) { 
    	int st=0,end=n-1;
    	long long max=arr[n-1]+1;
    	for(int i=0;i<n;i+=2){
    	    int c=arr[end--]%max;
    	    arr[i]=arr[i]+c*max;
    	    c=arr[st++]%max;
    	    arr[i+1]=arr[i+1]+c*max;
    	}
    	
    	for(int i=0;i<n;i++)
    	arr[i]/=max;
    }
